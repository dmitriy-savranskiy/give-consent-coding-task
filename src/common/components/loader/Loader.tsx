import { makeStyles } from '@material-ui/core';
import { tidLoader } from '../../testUtils/testIds';

const useStyles = makeStyles((theme) => ({
  loader: {
    ...theme.typography.body1,
  },
}));

export function Loader(): JSX.Element {
  const classes = useStyles();

  return (
    <span data-testid={tidLoader()} className={classes.loader}>
      Loading...
    </span>
  );
}
