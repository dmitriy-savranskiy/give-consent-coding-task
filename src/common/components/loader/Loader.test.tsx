import { create } from 'react-test-renderer';
import { Loader } from './Loader';

describe('Loader component', () => {
  test('should match snapshot', () => {
    const component = create(<Loader />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
