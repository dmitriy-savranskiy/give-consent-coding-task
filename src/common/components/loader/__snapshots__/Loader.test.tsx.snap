// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should match snapshot 1`] = `
<span
  className="makeStyles-loader-1"
  data-testid="LOADER"
>
  Loading...
</span>
`;
